core/src/main/java/uk/gov/gchq/magmacore/service/MagmaCoreService.java

```diff
@@ -345,6 +345,72 @@ public List<? extends Thing> findByPartialSignAndClassCaseSensitive(final String
 
     }
 
+    /**
+     * A case-insensitive search for entities in a specified class with a sign containing the given text.
+     *
+     * @param text        The String to search for.
+     * @param classIri    The IRI of the class that the entities should be a member_of.
+     * @param pointInTime When the entities should have the matching sign.
+     * @return A {@link List} of {@link Thing}.
+     */
+    public List<? extends Thing> findByPartialSignAndClassCaseInsensitive(final String text, final IRI classIri,
```

Comment:
I might suggest we remove `CaseInsensitive` from the end of these methods and queries. Is it reasonable to expect that the non-case sensitive queries are the _default_ methods to use?

```suggestion
    public List<? extends Thing> findByPartialSignAndClass(final String text, final IRI classIri,
```

Reply:
I preferred to be explicit with the query names since you have to do something special to make them case insensitive because it's not the default. The SPARQL is quite long for these so I think its helpful if the name gives an indication for what it does - same for the case sensitive ones as well even though they have default behaviour.